internal/services/appservice/helpers/function_app_schema.go

```diff
@@ -334,14 +334,323 @@ func SiteConfigSchemaLinuxFunctionApp() *pluginsdk.Schema {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: apimValidate.ApiID,
+					Description:  "The ID of the API Management API for this Windows Function App.",
+				},
+
+				"api_definition_url": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+					Description:  "The URL of the API definition that describes this Windows Function App.",
+				},
+
+				"app_command_line": {
+					Type:        pluginsdk.TypeString,
+					Optional:    true,
+					Description: "The program and any arguments used to launch this app via the command line. (Example `node myapp.js`).",
+				},
+
+				"app_scale_limit": {
+					Type:        pluginsdk.TypeInt,
+					Optional:    true,
+					Computed:    true,
+					Description: "The number of workers this function app can scale out to. Only applicable to apps on the Consumption and Premium plan.",
+					// TODO Validation?
+				},
+
+				"application_insights_key": {
+					Type:      pluginsdk.TypeString,
+					Optional:  true,
+					Sensitive: true,
+					RequiredWith: []string{
+						"site_config.0.application_insights_connection_string",
+					},
+					ValidateFunc: validation.StringIsNotEmpty,
+					Description:  "The Instrumentation Key for connecting the Windows Function App to Application Insights.",
+				},
+
+				"application_insights_connection_string": {
+					Type:      pluginsdk.TypeString,
+					Optional:  true,
+					Sensitive: true,
+					RequiredWith: []string{
+						"site_config.0.application_insights_key",
+					},
+					ValidateFunc: validation.StringIsNotEmpty,
+					Description:  "The Connection String for linking the Windows Function App to Application Insights.",
+				},
+
+				"application_stack": windowsFunctionAppStackSchema(),
+
+				"app_service_logs": FunctionAppAppServiceLogsSchema(),
+
+				"default_documents": {
+					Type:     pluginsdk.TypeList,
+					Optional: true,
+					Computed: true,
+					Elem: &pluginsdk.Schema{
+						Type: pluginsdk.TypeString,
+					},
+					Description: "Specifies a list of Default Documents for the Windows Web App.",
+				},
+
+				"elastic_instance_minimum": {
+					Type:        pluginsdk.TypeInt,
+					Optional:    true,
+					Computed:    true,
+					Description: "The number of minimum instances for this Windows Function App. Only affects apps on Elastic Premium plans.",
+				},
+
+				"http2_enabled": {
+					Type:        pluginsdk.TypeBool,
+					Optional:    true,
+					Default:     false,
+					Description: "Specifies if the http2 protocol should be enabled. Defaults to `false`.",
+				},
+
+				"ip_restriction": IpRestrictionSchema(),
+
+				"scm_use_main_ip_restriction": {
+					Type:        pluginsdk.TypeBool,
+					Optional:    true,
+					Default:     false,
+					Description: "Should the Windows Function App `ip_restriction` configuration be used for the SCM also.",
+				},
+
+				"scm_ip_restriction": IpRestrictionSchema(),
```

Comment:
should these properties conflict?

Reply:
The service accepts and updates this even if the `ScmIPSecurityRestrictionsUseMain` is `true`.  There may be use-cases where this is necessary?